core/arch/arm/plat-sam/platform_config.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2017 Timesys Corporation.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef PLATFORM_CONFIG_H
+#define PLATFORM_CONFIG_H
+
+#define STACK_ALIGNMENT       64
+
+#ifdef CFG_WITH_PAGER
+#error "Pager not supported for platform sama5d2"
+#endif
+#ifdef CFG_WITH_LPAE
+#error "LPAE not supported"
+#endif
+
+#define CONSOLE_UART_BASE     AT91C_BASE_UART1
+
+#define CFG_TEE_CORE_NB_CORE  1
+
+/*
+ * Everything is in DDR memory
+ * +------------------+
+ * |        | TA_RAM  |
+ * + TZDRAM +---------+
+ * |        | TEE_RAM |
+ * +------------------+
+ * |      SHMEM       |
+ * +------------------+
+ * |   Linux memory   |
+ * +--------+---------+
+ */
```

Comment:
So optee runs in low memory, Linux runs in high memory? To ARM32 Linux, I recall that there is limitation that linux image needs to be put in the lower 128M address space. If OP-TEE runs in low memory, you may need to reserve the lower 128M to OP-TEE and linux could not use that.